### DISCUSSION TITLE
Every single title in Korean shows up as ??????????

### DISCUSSION THREAD

**Discussion by thickmelon** [navidrome/navidrome#2941] [Q&A] [answered]
The app itself seems to have Korean support in the UI, but it somehow doesn't seem to be able to read a language that isn't in latin characters. Can this be patched....?

**Comment by metalheim** [comment]
I cannot confirm that Navidrome only works with latin characters!
Japanese, mandarin, cyrillic and also korean characters work well for me.
Here is an example file with korean chars in each tag and the file name:

![image](https://github.com/navidrome/navidrome/assets/44159957/1f06413a-2039-4951-ad5f-777349f92d1d)
![image](https://github.com/navidrome/navidrome/assets/44159957/2d62e44d-f799-40dc-9730-edcf1b29168f)

So my guess is that
- your file isnt properly encoded
- your file has invalid tags
- you use incompatible files (f.e. WAV files which are notorious for not haveing a proper text chunk encoding set or are tagged out-of-standard [f.e. with id3 tags])

Can you please share what kind of file (file format) and what tags you have?
Ideally show us a screenshot from a tagging program like mp3tag.

**Reply by thickmelon** [reply]
Should have made an update. Sorry.
I got it working.

I was working with flac files.

However, what I found weird was, after I used Picard to retag my files, I found Navidrome working with Korean again.

How does Navidrome extract titles exactly..?

**Reply by deluan** [reply, marked as answer]
> How does Navidrome extract titles exactly..?

It uses `taglib` by default but can be configured to use `ffmpeg`. You can check what tags it "sees" with the `inspect` command:

```
$ ./navidrome inspect --help                                                                                                                                                                
Show file tags as seen by Navidrome

Usage:
  navidrome inspect [files to inspect] [flags]

Flags:
  -x, --extractor string   extractor to use (ffmpeg or taglib, default: auto)
  -f, --format string      output format (pretty, toml, yaml, json, jsonindent) (default "pretty")
  -h, --help               help for inspect

Global Flags:
      --cachefolder string   folder to store cache data (transcoding, images...), needs write access
  -c, --configfile string    config file (default "./navidrome.toml")
      --datafolder string    folder to store application data (DB), needs write access (default ".")
  -l, --loglevel string      log level, possible values: error, info, debug, trace (default "info")
      --musicfolder string   folder where your music is stored (default "music")
```
